source/extensions/common/proxy_protocol/proxy_protocol_header.cc

```diff
@@ -72,8 +72,8 @@ void generateV2Header(const std::string& src_addr, const std::string& dst_addr,
         Network::Address::Ipv4Instance(src_addr, src_port).ip()->ipv4()->address();
     const auto net_dst_addr =
         Network::Address::Ipv4Instance(dst_addr, dst_port).ip()->ipv4()->address();
-    memcpy(addrs, &net_src_addr, 4);
-    memcpy(&addrs[4], &net_dst_addr, 4);
+    memcpy(addrs, &net_src_addr, 4);     // NOLINT(safe-memcpy)
```

Comment:
I think these all can be SAFE_MEMCPY.

Reply:
Sound like the right choice, though, since the buffer should concatenate the two addresses (src + dst), I will test MemBlockBuilder's appendData method first, just to see if it fits better to this use case ;-).